### PR TITLE
TextControl: Deprecate 36px default size

### PR DESCRIPTION
**Pull request by vipul0425**
Part of #65751 

## What?
Deprecate the 36px default size in TextControl.

## Testing Instructions

- Unit tests pass
- Storybook stories should not log console warnings
- All code snippets in documentation (JSDoc, Storybook, README) should have the __next40pxDefaultSize prop enabled

### CODE REVIEW

File: packages/components/src/text-control/index.tsx [hunk 1]
```diff
@@ -38,6 +39,13 @@ function UnforwardedTextControl(
 	const onChangeValue = ( event: ChangeEvent< HTMLInputElement > ) =>
 		onChange( event.target.value );
 
+	// Add Warning for deprecated 36px size.
```
Comment (mirka) [inline packages/components/src/text-control/index.tsx]: Comment seems unnecessary?

```suggestion
```

**Review by mirka — COMMENTED** [review]
Changes look good 👍 

In addition to this though, there are a lot of code snippets in Markdown files and Storybook stories across the repo that use TextControl, and all of them need the prop. Otherwise the devs referring to those snippets would be copy pasting deprecated code. (I added some clarification in the [original issue](https://github.com/WordPress/gutenberg/issues/65751)'s procedure checklist.)

**Comment by vipul0425** [comment]
Hi @mirka I have updated the markdown files and stories. 
Thanks 🙇

**Review by mirka — APPROVED** [review]
Awesome, thank you!

**Pull request merged**